whetstone/src/main/java/com/deliveryhero/whetstone/logging/GlobalAndroidComponentListener.kt

```diff
@@ -0,0 +1,15 @@
+package com.deliveryhero.whetstone.logging
+
+import java.lang.IllegalStateException
+
+public object GlobalAndroidComponentListener {
+    public var componentInjectionListener : ComponentInjectionListener? = null
+     set(value) {
+        if (componentInjectionListener != null && value != null) {
+            // Consider adding a wrapper around this callback if you need
+            // more than callback at same time
+            throw IllegalStateException("You are overriding and existing callback")
```

Comment:
do we really care about overriding existing callbacks? so generally it's fine to do `componentInjectionListener  = null; componentInjectionListener  = something`, but not directly?

Reply:
I was trying to make it more explicit. Consider someone has added the callback somewhere in one PR and another developer use it without being transparent. yeah it's still possible to make it null and that's somehow show the person know he is removing something.